### PR TITLE
*: add libgeos to more places

### LINKED ISSUE #61665: bazel: incorporate libgeos into tests that need it
At LEAST the following tests need to be able to find `libgeos` somewhere:

```
pkg/geo/geogfn:geogfn_test
pkg/geo/geoindex:geoindex_test
pkg/geo/geomfn:geomfn_test
pkg/geo/geos:geos_test
pkg/sql/opt/exec/execbuilder:execbuilder_test
pkg/ccl/logictestccl:logictestccl_test
pkg/sql/logictest:logictest_test
```

Presumably `acceptance` as well, and maybe others I haven't noticed yet.

### PR DESCRIPTION
**Pull request by otan**
See individual commits for more details.

Resolves #61665

### CODE REVIEW

**Comment by RaduBerinde** [comment]
Can you add it to these tests as well?

memo.TestStats
norm.TestNormRules
xform.TestExternal
xform.TestRules

I'm fixing them in #62109 but they still fail because of libgeos.

**Comment by otan** [comment]
done!

File: pkg/sql/BUILD.bazel [hunk 1]
```diff
@@ -489,7 +489,9 @@ go_test(
         "zone_config_test.go",
         "zone_test.go",
     ],
-    data = glob(["testdata/**"]),
+    data = glob(["testdata/**"]) + [
+        "@cockroach//c-deps:libgeos",
```
Comment (rickystewart) [inline pkg/sql/BUILD.bazel:493]: Why was this necessary here?

**Review by rickystewart — APPROVED** [review]

File: pkg/sql/BUILD.bazel (hunk 1 above)
Comment (otan) [reply pkg/sql/BUILD.bazel:493]: the sql logic tests will depend on this

**Comment by otan** [comment]
bors r=rickystewart
